Cython/Compiler/Pipeline.py

```diff
@@ -194,7 +194,6 @@ def create_pipeline(context, mode, exclude_classes=()):
         AutoCpdefFunctionDefinitions(context),
         RemoveUnreachableCode(context),
         ConstantFolding(),
-        FlattenInListTransform(),
```

Comment:
This is intentionally done early. Why do you want to move it?

Reply:
Because ctuple `in` and `not in` require type information as they can only compare compatible types, to match with python behavior.